scipy/stats/_multivariate.py

```diff
@@ -4275,3 +4277,448 @@ def rvs(self, size=1, random_state=None):
+        M, n = M[..., np.newaxis], n[..., np.newaxis]
+        output = n * m/M * (M-m)/M * (M-n)/(M-1)
+        return self._checkresult(output, mncond, np.nan)
+
+    def cov(self, m, n):
+        """
+        Covariance matrix of the Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        cov : array_like
+            The covariance matrix of the distribution
+        """
+        # see [1]_ for the formula and [2]_ for implementation
+        # cov( x_i,x_j ) = -n * (M-n)/(M-1) * (K_i*K_j) / (M**2)
+        M, m, n, _, _, mncond = self._process_parameters(m, n)
+        M = M[..., np.newaxis, np.newaxis]
+        n = n[..., np.newaxis, np.newaxis]
+        output = (-n * (M-n)/(M-1) / (M**2) *
+                  np.einsum("...i,...j->...ij", m, m))
+        dim = m.shape[-1]
+        M, n = M[..., 0, 0], n[..., 0, 0]
+        # diagonal entries need to be computed differently
+        for i in range(dim):
+            output[..., i, i] = (n * (M-n)/(M-1) *
+                                 m[..., i]*(M-m[..., i]) / (M**2))
+
+        return self._checkresult(output, mncond, np.nan)
+
+    def rvs(self, m, n, size=1, random_state=None):
+        """
+        Draw random samples from a Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+        size : integer or iterable of integers, optional
+            Number of samples to draw (default 1).
+        %(_doc_random_state)s
+
+        Returns
+        -------
+        rvs : array_like
+            Random variates of shape (`size`, `len(p)`)
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        M, m, n, _, _, mncond = self._process_parameters(m, n)
+
+        random_state = self._get_random_state(random_state)
+        # Generator is only available in numpy >= 1.17 while
+        # `multivariate_hypergeometric` distribution is only
+        # available in numpy >= 1.18
+        if (np.__version__ >= '1.18' and
+            isinstance(random_state, np.random.Generator)):  # noqa: E129
+            out = random_state.multivariate_hypergeometric(m, n, size)
+            return _squeeze_output(out)
+
+        if isinstance(size, int):
+            size = (size, )
+
+        rvs = np.empty(size + (m.shape[-1], ), dtype=m.dtype)
+        rem = M
+
+        # This sampler has been taken from numpy gh-13794
+        # https://github.com/numpy/numpy/pull/13794
+        for c in range(m.shape[-1] - 1):
+            rem = rem - m[..., c]
+            rvs[..., c] = ((n != 0) *
+                            random_state.hypergeometric(m[..., c], rem,
+                                                        n + (n == 0),
+                                                        size=size))
+            n = n - rvs[..., c]
+        rvs[..., m.shape[-1] - 1] = n
+
+        return _squeeze_output(rvs)
```

Comment:
This might be copied from other code, but I think this indiscriminate squeeze is a mistake.  If I pass in a 1-d array of length L for `m` and a scalar `n`, I expect a 1-d result with shape `(L,)`.  This should be true even if `L` is one.  So in this example,
```
In [18]: multivariate_hypergeom.rvs([10], 5)
Out[18]: 5
```
I expect the result to be `np.array([5])`, not the scalar `5`.  Actually, since the default `size` is 1, I expect the result to be `np.array([[5]])`.  That is, when `size` is an integer, I expect the result to have shape `(size, L)`, even if either `size` or `L` is 1.  The return shape is an invariant, with no indiscriminate squeezes to break that invariant.  I would also prefer the default size to be `None`, which means "return a single variate" instead of an array of variates, and the shape of the return value is `(L,)`.

Unfortunately, a few of the other multivariate distributions have set a precedent of using `size=1` as the default.  It looks like a good model for this and future PRs is `multinomial`, which behaves as I expect:
```
In [38]: multinomial.rvs(5, [0.1, 0.2, 0.7])  # Default size is None; return shape is (3,).
Out[38]: array([0, 1, 4])

In [39]: multinomial.rvs(5, [0.1, 0.2, 0.7], size=1)  # Explicit size=1; return shape is (1, 3).
Out[39]: array([[0, 3, 2]])

In [40]: multinomial.rvs(5, [0.1, 0.2, 0.7], size=4)  # Explicit size=4; return shape is (4, 3).
Out[40]: 
array([[1, 1, 3],
       [0, 2, 3],
       [1, 1, 3],
       [1, 0, 4]])
```
The rule applies even if the length of `p` is 1:
```
In [42]: multinomial.rvs(5, [1.0])  # Default size is None; return shape is (1,).
Out[42]: array([5])

In [43]: multinomial.rvs(5, [1.0], size=1)  # Explicit size=1; return shape is (1, 1).
Out[43]: array([[5]])

In [44]: multinomial.rvs(5, [1.0], size=4)  # Explicit size=4; return shape is (4, 1).
Out[44]: 
array([[5],
       [5],
       [5],
       [5]])
```
This is also how the random distributions in NumPy work.  For example, here's `multivariate_normal` in NumPy:
```
In [61]: rng = np.random.default_rng(998877665544332211)

In [62]: rng.multivariate_normal(mean=[0, 100], cov=[[4, 1], [1, 3]])
Out[62]: array([ 0.67523517, 98.65704622])

In [63]: rng.multivariate_normal(mean=[0, 100], cov=[[4, 1], [1, 3]], size=1)
Out[63]: array([[  1.17072195, 102.50028863]])

In [64]: rng.multivariate_normal(mean=[0, 100], cov=[[4, 1], [1, 3]], size=4)
Out[64]: 
array([[ 2.47625207e+00,  1.02905478e+02],
       [ 3.13979220e+00,  9.97893792e+01],
       [-7.02967469e-02,  1.03507684e+02],
       [ 2.08209013e+00,  1.00102516e+02]])
```

Reply:
Tried to resolve this. Can you please check if this is OK for your input instances? I have tested locally and it is compatible with numpy. Do we need tests for this?